### PR TITLE
Display samples for workflow execution (ENHC0010030)

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create a new project
2. Add a sample to the project and upload sequence files to the sample.
3. Go back to your new project samples page and select the new sample and launch a pipeline for it.
4. Go to workflow executions and select the new one.
5. Select the tab for `Samplesheet`
6. Files should be displayed as above.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/system/workflow_executions_test.rb [hunk 1]
```diff
@@ -342,6 +342,19 @@ class WorkflowExecutionsTest < ApplicationSystemTestCase
     assert_selector 'div.random_seed-param > input[value="1"]'
   end
```
Comment (deepsidhu85) [inline test/system/workflow_executions_test.rb:344]: Can I also get you to add a test for a samplesheet with a fastq_1 and fastq_2 input

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This is looking good. Other than the update required to `_samplesheet.html.erb` to only output the `puid` and `name` of the attachment if it was provided in the samplesheet when the pipeline was ran, just one comment below

File: test/system/workflow_executions_test.rb [hunk 2]
```diff
@@ -342,6 +344,34 @@ class WorkflowExecutionsTest < ApplicationSystemTestCase
...
+    click_on I18n.t('workflow_executions.show.tabs.samplesheet')
+
+    assert_selector 'table tbody tr', count: 1
+    assert_text 'INXT_SAM_AAAAAAAAAA'
+    assert_text 'INXT_ATT_AAAAAAAAAA'
+    assert_text 'test_file.fastq'
+    assert_text 'INXT_ATT_AAAAAAAAAB'
+    assert_text 'test_file_A.fastq'
+  end
+
```
Comment (JeffreyThiessen) [inline test/system/workflow_executions_test.rb:374]: Please add a test for a workflow execution with 0 samples.

File: app/controllers/concerns/workflow_execution_actions.rb [hunk 3]
```diff
@@ -98,4 +109,33 @@ def set_default_sort
   def redirect_path
     raise NotImplementedError
   end
+
+  def format_samplesheet_params
+    @samplesheet_headers = @workflow_execution.samples_workflow_executions.first.samplesheet_params.keys
```
Comment (JeffreyThiessen) [inline app/controllers/concerns/workflow_execution_actions.rb:114]: This line needs safe access, if the workflow execution has 0 samples this throws a undefined method exception.

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Overall it looks good. Just one edge case issue.

After launching a workflow execution with 0 samples (not sure if this is intended in the first place), when clicking on the samplesheet tab, the application throws a error to the user because of an undefined method.

image

File: test/system/workflow_executions_test.rb (hunk 1 above)
Comment (joshsadam) [reply test/system/workflow_executions_test.rb:346]: Added in 8d54e7c

File: test/system/workflow_executions_test.rb (hunk 2 above)
Comment (joshsadam) [reply test/system/workflow_executions_test.rb:374]: Now that you cannot launch a workflow without any samples this is not required.

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Review by JeffreyThiessen — APPROVED** [review]
:+1:

**Pull request merged**